Enable pdc

leandro-deveikis opened the pull request.

baldm0mma reviewed (approved):
LGTM, but I'd love to get another pair of eyes on it also; I'll repost in our BI channel.

leandro-deveikis commented:
I have also tested it with the code in `main` and it seems that is the same, every time that you hit save, it creates a new BQ connection even if the configuration has not changed.

wbrowne commented:
>I have just tested it and it seems that you are correct, every time that I hit "Save and test" it creates a new connection (checking the logs) even if the parameters are the same. Most of the plugins I worked on so far just create a new connection when the config is updated

Save and Test will always create a new plugin instance under the hood because the timestamp of the datasource within Grafana has changed - see [here](https://github.com/grafana/grafana-plugin-sdk-go/blob/main/backend/datasource/instance_provider.go#L70-L72).

I'm just not sure if the changes other than to the HTTP client and the UI are necessary here. I'd be happy to test that, but I reckon you're in a better position to do so 😊

leandro-deveikis commented:
Thanks will, you are right. 

I will undo the changes to the connection cache, leaving it as it was before (and leaving that task of checking if it is necessary for another ticket, to be created), and test it again.

leandro-deveikis commented:
@wbrowne I have reverted the changes to the cache key and tested it. It works as expected, every time you save a new config, the connection is recreated. Right now the cache is useless, but the same code as before. Is that ok?

@baldm0mma should we wait for someone else to review before merging?


I have also updated the PR description. 
Thanks both of you!

wbrowne commented:
>Right now the cache is useless

The cache will be used when you're using the plugin as normal (e.g. querying) so it still does serve some purpose. As I mention, `Save & Test` for a data source connection will actually update the fields under the hood and trigger a recreation of a plugin instance. Though, if you've seen something that contradicts that, please let me know!

Glad to hear it works as expected 😄

wbrowne reviewed (approved):
LGTM

leandro-deveikis commented:
> The cache will be used when you're using the plugin as normal (e.g. querying) so it still does serve some purpose. 

Yeah, you are right about that. I meant that having a cache is not worth it as it always has only one instance of the connection in it at any time. Most datasources projects I have seen store the connection as an attribute in a struct.

leandro-deveikis commented:
Thank you both!! Merging and closing!
